### PR TITLE
string: add the rest of the functions

### PR DESCRIPTION
**Pull request by davidvgalbraith**
- [x] String.fromCharCode
- [x] String.charAt
- [x] String.charCodeAt
- [x] String.match
- [x] String.trim

relates to https://github.com/juttle/juttle/issues/58

### CODE REVIEW

File: lib/compiler/semantic.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ class SemanticPass {
                 /* Functions (sorted alphabetically) */
                 length: { type: 'function', exported: true, uname: 'juttle.modules.string.length', arg_count: 1             },
                 concat: { type: 'function', exported: true, uname: 'juttle.modules.string.concat', arg_count: [0, Infinity] },
+                fromCharCode: { type: 'function', exported: true, uname: 'juttle.modules.string.fromCharCode', arg_count: [1, Infinity]},
```
Comment (rlgomes) [inline lib/compiler/semantic.js]: Why not start from 0 ? Its a valid character code.

File: test/runtime/specs/juttle-spec/modules/string/from-char-code.spec.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+The `String.fromCharCode` function
+============================
+
+Produces an error when passed an argument of incorrect type
+-----------------------------------------------------------
+
+### Juttle
+
+    emit -from Date.new(0) -limit 1 | put result = String.fromCharCode(null) | view result
```
Comment (rlgomes) [inline test/runtime/specs/juttle-spec/modules/string/from-char-code.spec.md]: I've seen this in a few other `juttle-spec` tests but is there a good reason why we tend to enumerate the negative tests and never add a positive test ? At least that way the coverage of that one line 27 above would happen unlike the current situation of this commit where that one line would be another "uncovered" line.

File: lib/compiler/semantic.js (hunk 1 above)
Comment (davidvgalbraith) [inline lib/compiler/semantic.js]: What? `arg_count` is the number of arguments, and `String.fromCharCode()` doesn't make sense.

File: test/runtime/specs/juttle-spec/modules/string/from-char-code.spec.md (hunk 2 above)
Comment (davidvgalbraith) [inline test/runtime/specs/juttle-spec/modules/string/from-char-code.spec.md]: What? I have a `fromCharCode works as expected` test right there on line 15.

File: lib/compiler/semantic.js (hunk 1 above)
Comment (davidvgalbraith) [inline lib/compiler/semantic.js]: I guess it makes sense, we'd just return empty string.

**Comment by davidvgalbraith** [comment]
@dmajda @bkutil

**Comment by dmajda** [comment]
+1

**Pull request merged**